java8/src/main/java/cucumber/runtime/java8/ConstantPoolTypeIntrospector.java

```diff
@@ -42,21 +42,16 @@
         return typeArguments;
     }
 
-    private String getLambdaTypeString(ConstantPool constantPool) {
-        int size = constantPool.getSize();
-        String[] memberRef = null;
-
-        // find last element in constantPool with valid memberRef
-        // - previously always at size-2 index but changed with JDK 1.8.0_60
-        for (int i = size - 1; i > -1; i--) {
+    String getLambdaTypeString(ConstantPool constantPool) {
+        int index = constantPool.getSize();
+        while (index-- > 0) {
```

Comment:
Personally I find a `for` loop clearer when iterate using an index. I try to avoid side effects when checking conditions